compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2278,6 +2278,8 @@ impl BinOp {
 pub enum NullOp {
     /// Returns the size of a value of that type
     SizeOf,
+    /// Returns the minimum alignment of a type
+    AlignOf,
```

Comment:
Maybe call this `MinAlignOf` to distinguish it from the preferred alignment?

Reply:
Given that preferred alignment is never used in the library and hardly used in the compiler (in fact the only usage of preferred alignment is in cg_clif), and the name we expose `min_align_of` to the user is `align_of`, I think the name `align_of/AlignOf` is not ambiguous. I doubt if we'll ever add a `PrefAlignOf` nullary operator.

Reply:
Huh, indeed. Should we remove preferred alignment then if it isn't used anyway?

Reply:
> Should we remove preferred alignment then if it isn't used anyway?

Yeah it might be time to remove that intrinsic... @eddyb or do you think that will ever be useful again in the future?